edgexfoundry/go-mod-messaging
# feat: Enable use of CleanSession MQTT option

closes #111









## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I have opened a PR for the related docs change (if not, why?) Documented in README


## Testing Instructions

1. Clone edgex-compose
2. cd compose-builder
3. run `make no-secty mqtt-buss`
4. Clone the branch for this PR 
5. Add following at line 295 of `internal/pkg/mqtt/client.go`
    ```
	fmt.Printf("Clean Session is '%v'\n", clientOptions.CleanSession)
    ```
6. Clone app-functions-sdk-go
7.  Add the following to top level go.mod
   ```
replace (
	github.com/edgexfoundry/go-mod-messaging/v2 => ../MODS/go-mod-messaging
)
   ```
8. Add following to `app-service-template/go.mod`
```
replace (
	github.com/edgexfoundry/app-functions-sdk-go/v2 => ../
	github.com/edgexfoundry/go-mod-messaging/v2 => ../../MODS/go-mod-messaging
)
```
9. cd to app-service-template folder and run `make build`
10. Edit res/configuration.toml 
   - uncomment Trigger configuration for `mqtt` and Comment out Trigger configuration for `redis`
10. run `EDGEX_SECURITY_SECRET_STORE=false ./new-app-service`
11. Verify the line `Clean Session is 'true'` is logged. Note value is TRUE by default.
12. Hit CTRL-C to stop app.
12. Add `CleanSession   = "false"` to `[Trigger.EdgexMessageBus.Optional]` section in TOML
13. run `EDGEX_SECURITY_SECRET_STORE=false ./new-app-service`
11. Verify the line `Clean Session is 'false'` is logged. Note value is now FALSE.
